op-chain-ops/cmd/receipt-reference-builder/main.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"os"
+	"time"
+
+	"github.com/mattn/go-isatty"
+	"github.com/urfave/cli/v2"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+
+	opservice "github.com/ethereum-optimism/optimism/op-service"
+)
+
+const EnvPrefix = "OP_CHAIN_OPS_RECEIPT_REFERENCE_BUILDER"
+
+var (
+	StartFlag = &cli.Uint64Flag{
+		Name:    "start",
+		Usage:   "the first block to include in data collection. INCLUSIVE",
+		EnvVars: opservice.PrefixEnvVar(EnvPrefix, "START"),
+	}
+	EndFlag = &cli.Uint64Flag{
+		Name:    "end",
+		Usage:   "the last block of the collection range. EXCLUSIVE",
+		EnvVars: opservice.PrefixEnvVar(EnvPrefix, "END"),
+	}
+	RPCURLFlag = &cli.StringFlag{
+		Name:    "rpc-url",
+		Usage:   "RPC URL to connect to",
+		EnvVars: opservice.PrefixEnvVar(EnvPrefix, "RPC_URL"),
+	}
+	BackoffFlag = &cli.DurationFlag{
+		Name:    "backoff",
+		Value:   30 * time.Second,
+		Usage:   "how long to wait when a worker errors before retrying",
+		EnvVars: opservice.PrefixEnvVar(EnvPrefix, "BACKOFF"),
+	}
+	WorkerFlag = &cli.Uint64Flag{
+		Name:    "workers",
+		Value:   1,
+		Usage:   "how many workers to use to fetch txs",
+		EnvVars: opservice.PrefixEnvVar(EnvPrefix, "WORKERS"),
+	}
+	BatchSizeFlag = &cli.Uint64Flag{
+		Name:    "batch-size",
+		Value:   50,
+		Usage:   "how many blocks to batch together for each worker",
+		EnvVars: opservice.PrefixEnvVar(EnvPrefix, "BATCH_SIZE"),
+	}
+	OutputFlag = &cli.StringFlag{
+		Name:    "output",
+		Aliases: []string{"o"},
+		Usage:   "the file to write the results to",
+		EnvVars: opservice.PrefixEnvVar(EnvPrefix, "OUTPUT"),
+	}
+	FilesFlag = &cli.StringSliceFlag{
+		Name:    "files",
+		Aliases: []string{"f"},
+		Usage:   "the set of files to merge",
+	}
+	InputFormatFlag = &cli.StringFlag{
+		Name:    "input-format",
+		Aliases: []string{"if"},
+		Value:   "json",
+		Usage:   "the format to read aggregate files: json, gob",
+		EnvVars: opservice.PrefixEnvVar(EnvPrefix, "OUTPUT"),
+	}
+	OutputFormatFlag = &cli.StringFlag{
+		Name:    "output-format",
+		Aliases: []string{"of"},
+		Value:   "json",
+		Usage:   "the format to write the results in. Options: json, gob",
+		EnvVars: opservice.PrefixEnvVar(EnvPrefix, "OUTPUT"),
```

Comment:
ditto
edit: TBH we could drop all env vars except the RPC URL.

Reply:
Yeah gonna remove all but RPC_URL. Was trying to match format but tbh these aren't things that will ever be loaded as env